OpenRA.Game/Actor.cs

```diff
@@ -378,17 +379,21 @@ public bool IsTargetableBy(Actor byActor)
 
 		#region Scripting interface
 
-		Lazy<ScriptActorInterface> luaInterface;
+		//Lazy<ScriptActorInterface> luaInterface;
+		Dictionary<LuaRuntime, Lazy<ScriptActorInterface>> luaInterfaces = new Dictionary<LuaRuntime, Lazy<ScriptActorInterface>>();
 		public void OnScriptBind(ScriptContext context)
 		{
-			if (luaInterface == null)
-				luaInterface = Exts.Lazy(() => new ScriptActorInterface(context, this));
+			if (!luaInterfaces.ContainsKey(context.Runtime))
+			{
+				var luaInterface = Exts.Lazy(() => new ScriptActorInterface(context, this));
+				luaInterfaces.Add(context.Runtime, luaInterface);
+			}
 		}
 
 		public LuaValue this[LuaRuntime runtime, LuaValue keyValue]
 		{
-			get { return luaInterface.Value[runtime, keyValue]; }
-			set { luaInterface.Value[runtime, keyValue] = value; }
+			get { return luaInterfaces[runtime].Value[runtime, keyValue]; }
```

Comment:
This is already passing the runtime in to .Value, which might indicate that the multi-runtime awareness would better live there.